packages/yew-router/src/lib.rs

```diff
@@ -80,6 +80,7 @@ pub mod utils;
 pub use routable::{AnyRoute, Routable};
 pub use router::{BrowserRouter, HashRouter, Router};
 pub use switch::Switch;
+pub use urlencoding::encode as encode_for_url;
```

Comment:
I think this should be exposed via "macro_herlpers.rs".

In addition, it might worth to test `js_sys::encode_uri_component` and prefer it under wasm targets and see if using the function provided by `js_sys` can avoid the size increase.

Reply:
I agree that this should be exposed via macro_helpers and certainly not be a part of a public API. @Jaffa-Cakes please move it over to that file.

I don't think `js_sys::encode_uri_component` is worth it here though. The binary size increase is minimal and the JS round trip: (UTF8 (rust memory)  -> UTF16 (JS memory) ->  UTF8 (rust memory)) may be too costly

Reply:
I think the primary performance hit when routes are switched comes from rendering components of the new page after the URL switch. I feel it might be worth to avoid the size increase here as we are looking at 5% increase of example sizes for the next release based on upcoming pull requests (e.g.: #3169, #3050) and the other size increases are not likely avoidable.

But I don't have a high preference over either choice, so I am fine with proceeding the Rust dependency as well.